src/lib/config/config.go

```diff
@@ -126,3 +133,109 @@ func Load(ctx context.Context) error {
 func Upload(cfg map[string]interface{}) error {
 	return defaultMgr().UpdateConfig(orm.Context(), cfg)
 }
+
+// AssignByTag set the value of config model by tags, if it exist in cfgs, it is updated, if it comes from cfgManager, updated is false
+// before using this function, the field of the model should have config tag or json tag, for example
+// type GroupConf struct {
+//	BaseDN              string `json:"ldap_group_base_dn,omitempty"`
+//	Filter              string `config:"ldap_group_search_filter" json:"ldap_group_filter,omitempty"`
```

Comment:
it seems the "config" and "validate" annotations are only needed when you want custom validation, and there's only one case in whole Harbor?

Reply:
config, validate only required when need to use the validation lib